app/src/main/java/com/github/se/icebreakrr/ui/authentication/SignIn.kt

```diff
@@ -0,0 +1,235 @@
+  val verticalPadding =
+      when {
+        screenHeight > 800 -> 150.dp
+        else -> 100.dp
+      }
+
+  val verticalSpacing =
+      when {
+        screenHeight > 800 -> 250.dp
+        else -> 150.dp
+      }
+
+  val launcher =
+      rememberFirebaseAuthLauncher(
+          onAuthComplete = { result ->
+            user = result.user
+            Toast.makeText(context, "Login successful!", Toast.LENGTH_LONG).show()
+            navigationActions.navigateTo(TopLevelDestinations.AROUND_YOU)
+          },
+          onAuthError = { user = null })
+
+  // Define a linear gradient with the provided colors
+  val gradientBrush =
+      Brush.linearGradient(
+          colors =
+              listOf(
+                  Color(0xFF1FAEF0), // Light blue
+                  Color(0xFF1C9EDA), // Mid blue
+                  Color(0xFF12648A) // Dark blue
+                  ))
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize().testTag("loginScreen"),
+      content = { paddingValues ->
+        Column(
+            modifier =
+                Modifier.background(brush = gradientBrush)
+                    .fillMaxSize()
+                    .padding(vertical = verticalPadding)
+                    .padding(paddingValues),
+            verticalArrangement = Arrangement.spacedBy(verticalSpacing, Alignment.Top),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+          Text(
+              modifier = Modifier.testTag("loginTitle"),
+              text = "IceBreakrr",
+              style =
+                  TextStyle(
+                      fontSize = 64.sp,
+                      lineHeight = 25.07.sp,
+                      fontWeight = FontWeight(500),
+                      color = Color(0xFFFFFFFF),
+                      textAlign = TextAlign.Center,
+                      letterSpacing = 0.37.sp,
+                  ))
+
+          // Authenticate With Google Button
+          GoogleSignInButton(
+              onSignInClick = {
+
+                // If in test mode, simulate a logged-in user
+                if (isTesting) {
+                  navigationActions.navigateTo(TopLevelDestinations.AROUND_YOU)
+                } else {
+                  val gso =
+                      GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                          .requestIdToken(token)
+                          .requestEmail()
+                          .build()
+                  val googleSignInClient = GoogleSignIn.getClient(context, gso)
+                  launcher.launch(googleSignInClient.signInIntent)
+                }
+              })
+        }
+      })
+}
+
+/**
+ * Composable function to display the Google Sign-In button. It is a custom-styled button that
+ * triggers the Google Sign-In flow when clicked.
+ *
+ * @param onSignInClick Callback invoked when the button is clicked.
+ */
+@Composable
+fun GoogleSignInButton(onSignInClick: () -> Unit) {
```

Comment:
`onSignInClick: () -> Unit`

Nitpick: could be simply be replaced by `onClick: () -> Unit` for consistency with other composable (it's ok to have the same parameter name in your signature as in another composable's signature)

Reply:
Yes I agree mate!